signac/contrib/project.py

```diff
@@ -778,7 +788,7 @@ def write_statepoints(self, statepoints=None, fn=None, indent=2):
             tmp = dict()
         if statepoints is None:
             job_ids = self._job_dirs()
-            _cache = {_id: self.get_statepoint(_id) for _id in job_ids}
+            _cache = {_id: self.open_job(id=_id).sp for _id in job_ids}
```

Comment:
Let's change `sp` to `statepoint()`. `sp` and `statepoint` are aliased attributes (they're equivalent), but we need to use `dict`s in the cache instead of synced attribute dictionaries. If you call a synced dict (with `()`), it returns a standard `dict`.
```suggestion
            _cache = {_id: self.open_job(id=_id).statepoint() for _id in job_ids}
```